### PR TITLE
Deploy to heroku

### LINKED ISSUE #85: Deploy to Heroku

### PR DESCRIPTION
**Pull request by shizuko-akamoto**
Closes #85 

### It's time to do continuous deployment on master!!

TL:DR of deployment process:
1. client install + client build (creates a new `/build` folder)
2. server install + server build (compiles all ts into js and copies files required for prod into `/dist` folder. ie. PUBLIC_KEY file)
3. copy client `/build` into server `/dist`

Once this PR is in master, I guess we all have to be careful not to break the build... We can kind of simulate prod by:
1. run `yarn workspace client build`
2. run `yarn workspace server compile`
3. manually copy client's `/build` into server's `/dist`
4. run `yarn start`
5. go to `localhost:7000`. In prod mode, we serve our react app as static files from express server.
**IMPORTANT NOTE**: make sure in your server side's `.env`, you have `NODE_ENV=production`.

### CODE REVIEW

File: packages/server/server/common/passport/passportUtils.ts [hunk 1]
```diff
@@ -57,9 +59,18 @@ export function signJwt(user: User): Promise<string> {
     };
 
     return new Promise<string>((resolve, reject) => {
-        Jwt.sign(JwtPayload, process.env.PRIVATE_KEY, JwtOptions, (err, token) => {
-            if (err) return reject(err);
-            resolve(token);
-        });
+        Jwt.sign(
+            JwtPayload,
+            process.env.PRIVATE_KEY.replace(/\\n/g, '\n'),
```
Comment (knox153) [inline packages/server/server/common/passport/passportUtils.ts:64]: this must be annoying to deal with 😅

File: packages/server/server/common/server.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ const exit = process.exit;
 export default class ExpressServer {
     private routes: (app: Application) => void;
     constructor() {
-        const root = path.normalize(__dirname + '/../..');
+        const root = path.normalize(__dirname + '/../../..');
```
Comment (knox153) [inline packages/server/server/common/server.ts:23]: It is a bit confusing how this points to a different directory after we build/compile, maybe we can document this haha

File: package.json [hunk 3]
```diff
@@ -45,5 +53,9 @@
     "husky": "^4.2.5",
     "lint-staged": "^10.2.11",
     "prettier": "^2.0.5"
+  },
+  "engines": {
+    "node": "12.x",
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply package.json:58]: 🤣

File: packages/server/server/common/passport/passportUtils.ts (hunk 1 above)
Comment (shizuko-akamoto) [reply packages/server/server/common/passport/passportUtils.ts:64]: Nah this one was still ok, the logging gave pretty descriptive errors. I was stuck more last time i tried deploying (with the catch all * for app.get hahaha)

File: packages/server/server/common/server.ts (hunk 2 above)
Comment (shizuko-akamoto) [reply packages/server/server/common/server.ts:23]: Sure!
